django/contrib/admin/views/autocomplete.py

```diff
@@ -15,6 +17,34 @@ def get(self, request, *args, **kwargs):
             pagination: {more: true}
         }
         """
+        self.term = request.GET.get('term', '')
+
+        try:
+            app_label = request.GET['app_label']
+            model_name = request.GET['model_name']
+            field_name = request.GET['field_name']
+        except KeyError:
+            return JsonResponse({'error': '403 Forbidden'}, status=403)
+
+        try:
+            source_model = apps.get_model(app_label, model_name)
+        except LookupError:
+            return JsonResponse({'error': '403 Forbidden'}, status=403)
+
+        try:
+            self.source_field = source_model._meta.get_field(field_name)
+        except FieldDoesNotExist:
+            return JsonResponse({'error': '403 Forbidden'}, status=403)
+
+        try:
+            remote_model = self.source_field.remote_field.model
+        except AttributeError:
+            return JsonResponse({'error': '403 Forbidden'}, status=403)
+        try:
+            self.model_admin = self.admin_site._registry[remote_model]
+        except KeyError:
+            return JsonResponse({'error': '403 Forbidden'}, status=403)
```

Comment:
Why 403 instead of 404? If there is a good reason, maybe a commend would be helpful.

Reply:
How so, all errors lead to a 403-response, not a 404. Therefore, this seems consistent.

Reply:
Sorry if my comment was misleading. My point is that **all** of these should be 404 because 403 is mostly for permission checks.

If you look at the old code there were two checks: One about `get_search_fields` and one about permissions. The former used 404 and the latter 403.